src/Microsoft.AspNet.Routing/Template/RouteTemplate.cs

```diff
@@ -39,7 +39,7 @@ public RouteTemplate(string template, List<TemplateSegment> segments)
             }
         }
 
-        public string Template { get; }
+        public string TemplateText { get; }
```

Comment:
That's a good reason, but now you have a breaking change between MVC and routing so you have to check these in together :+1: